### ISSUE TITLE
mypy doesn't seem to enforce instance attributes Dict types

### ISSUE THREAD

**Issue by fpagnoux** [python/mypy#6986]
I would expect the following snippet to raise an error when checked by mypy, yet nothing is detected.

```py
class Test:
    def __init__(self):
        self.data: Dict[str, int] = {}

    def get(self, key) -> str:  # inconsistent with self.data type, should be int
        return self.data[key]
```

I'm using Python 3.7.0  and mypy 0.701
I'm still seeing the issue on `master`
I'm only using the flag  `ignore_missing_imports = True` 

Am I doing something wrong?

**Comment by ilevkivskyi** [comment]
This is a funny one. The problem is that you don't import `Dict`. Moreover `__init__` doesn't have any annotations, so errors aren't reported there, see PEP 484.

You can use `--disallow-untyped-defs` to avoid such mistakes in future, or alternatively `--warn-return-any` to warn you when type checking passes because the thing you return is `Any`.
